Index: RedmineManager.cs
===================================================================
--- RedmineManager.cs	(revision 52)
+++ RedmineManager.cs	(working copy)
@@ -171,6 +171,7 @@
                 {
                     using (var xmlReader = new XmlTextReader(text))
                     {
+                        xmlReader.WhitespaceHandling = WhitespaceHandling.None;
                         xmlReader.Read();
                         xmlReader.Read();
                         return xmlReader.ReadElementContentAsCollection<T>();
