Index: Types/TimeEntry.cs
===================================================================
--- Types/TimeEntry.cs	(revision 114)
+++ Types/TimeEntry.cs	(working copy)
@@ -126,8 +126,10 @@
                     case "id": Id = reader.ReadElementContentAsInt(); break;
 
                     case "issue_id": Issue = new IdentifiableName(reader); break;
+                    case "issue": Issue = new IdentifiableName(reader); break;
 
                     case "project_id": Project = new IdentifiableName(reader); break;
+                    case "project": Project = new IdentifiableName(reader); break;
 
                     case "spent_on": SpentOn = reader.ReadElementContentAsNullableDateTime(); break;
 
@@ -136,6 +138,7 @@
                     case "hours": Hours = reader.ReadElementContentAsDecimal(); break;
 
                     case "activity_id": Activity = new IdentifiableName(reader); break;
+                    case "activity": Activity = new IdentifiableName(reader); break;
 
                     case "comments": Comments = reader.ReadElementContentAsString(); break;
 
