Index: Types/MembershipRole.cs
===================================================================
--- Types/MembershipRole.cs	(revision 110)
+++ Types/MembershipRole.cs	(working copy)
@@ -44,6 +44,7 @@
             Id = Convert.ToInt32(reader.GetAttribute("id"));
             Name = reader.GetAttribute("name");
             Inherited = reader.ReadAttributeAsBoolean("inherited");
+            reader.Read();
         }
 
         public override void WriteXml(XmlWriter writer) { }
